### PR TITLE
 Add in isValidCalendarDate function 

### LINKED ISSUE #57: February cannot be more than 29 except for leap years

### PR DESCRIPTION
**Pull request by Yiheng0410**

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Date.java [hunk 1]
```diff
@@ -48,4 +48,16 @@ public boolean equals(Object other) {
...
         return date.matches(DATE_VALIDATION_REGEX);
     }
+
+    /**
+     * Returns true if a given string is a valid calendar date.
+     */
+    public static boolean isValidCalendarDate(String date) {
+        int days = Integer.parseInt(date.substring(0, 2));
+        int month = Integer.parseInt(date.substring(3, 5));
+        if (days > 28 && month == 2 ) {
```
Comment (Impala36) [inline src/main/java/seedu/address/model/person/Date.java]: This should also check for leap years, it would be 29 days instead.

**Review by Impala36 — CHANGES_REQUESTED** [review]
The condition should also check whether the year is a leap year.

File: src/main/java/seedu/address/model/person/Date.java (hunk 1 above)
Comment (Yiheng0410) [reply src/main/java/seedu/address/model/person/Date.java]: done

File: src/test/java/seedu/address/model/person/DateTest.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        // valid date
+        assertTrue(Date.isValidDate("12-12-2020"));
+        assertTrue(Date.isValidDate("22-02-2021"));
+    }
+
+    @Test
+    public void isValidCalendarDate() {
+        assertFalse(Date.isValidCalendarDate("29-02-2021")); // day in feb is bigger than 28
+        assertFalse(Date.isValidCalendarDate("30-02-2020")); // day in feb is bigger than 28
+        assertFalse(Date.isValidCalendarDate("30-02-2020")); // day in feb is bigger than 28
```
Comment (Impala36) [inline src/test/java/seedu/address/model/person/DateTest.java:44]: Line 43 and 44 are the same.

**Review by Impala36 — APPROVED** [review]

**Pull request merged**